### PR TITLE
Refactor package, file context and ProofFile execution

### PR DESCRIPTION
**Pull request by pcarrott**
- Reorganize directory structure to have a main `coqpyt` folder
- Automate changes to README with workflow
- Extract context management from CoqFile to FileContext
- Implement `exec` in ProofFile for interactive access on proofs
- Add optional `program` attribute in ProofTerm for obligation proofs

### CODE REVIEW

File: .github/workflows/readme.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Embed code in README
+
+on:
+  push:
+    paths:
+      - "examples/readme.py"
+      - "README.md"
```
[the remark this answers is not in the document]
Comment (pcarrott) [reply .github/workflows/readme.yml:7]: I agree. Will change

File: coqpyt/coq/proof_file.py [hunk 2]
```diff
@@ -469,12 +463,64 @@ def proofs(self) -> List[ProofTerm]:
         """
         return self.__proofs
 
-    def add_step(self, step_text: str, previous_step_index: int):
+    @property
+    def open_proofs(self) -> List[ProofTerm]:
+        """Gets all the proofs in the file and their corresponding steps.
```
[the remark this answers is not in the document]
Comment (pcarrott) [reply coqpyt/coq/proof_file.py:468]: I simply copied the code from `def proofs` without reading... Will change

File: coqpyt/coq/context.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        elif expr[0] == "VernacDeclareModuleType":
+            self.__segments.push(expr[1]["v"][1], SegmentType.MODULE_TYPE)
+        elif expr[0] == "VernacBeginSection":
+            self.__segments.push(expr[1]["v"][1], SegmentType.SECTION)
+        # HACK: We ignore terms inside a Module Type since they can't be used outside
+        # and should be overriden.
+        elif not self.in_module_type:
+            self.__add_terms(step, expr)
+
+    def expr(self, step: Step) -> Optional[List]:
```
[the remark this answers is not in the document]
Comment (Nfsaavedra) [reply coqpyt/coq/context.py:273]: Alright, let's go with that. In the future probably what we want is a class that deals with the AST, but for now this is OK.

File: coqpyt/coq/context.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+            and el[1][1][0] == "ExtraArg"
+        ):
+            if el[1][1][1] == "identref":
+                return el[2][0][1][1]
+            elif el[1][1][1] == "ident":
+                return el[2][1]
+        return None
+
+    @staticmethod
+    def get_v(el: List) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (Nfsaavedra) [reply coqpyt/coq/context.py:376]: You can make `get_v` private. Consistently wrong doesn't seem like a good principle to follow. But everything else makes sense. You don't need to add doc strings to these methods if we want to refactor them in the near future. But maybe we should add a comment stating the plan for those functions.

File: coqpyt/coq/proof_file.py [hunk 5]
```diff
@@ -221,214 +222,205 @@ def __init__(
...
-            self.__program_context: Dict[str, Tuple[Term, List]] = {}
-            self.__proofs = self.__get_proofs()
         except Exception as e:
             self.close()
             raise e
 
+        self.__program_context: Dict[str, Tuple[Term, List[Term]]] = {}
+        self.__proofs: List[ProofTerm] = []
+        self.__open_proofs: List[
+            Tuple[Term, List[Term], List[ProofStep], Optional[Term]]
```
[the remark this answers is not in the document]
Comment (Nfsaavedra) [reply coqpyt/coq/proof_file.py:237]: Alright, sounds like a good plan

**Review by Nfsaavedra — APPROVED** [review]
LGTM! When the tests finish you can merge :rocket:
https://www.youtube.com/watch?v=iik25wqIuFo

**Pull request merged**